api/server/server.go

```diff
@@ -66,26 +65,22 @@ func (s *Server) makeHTTPHandler(handler httputils.APIFunc, operation string) ht
 }
 
 // CreateMux returns a new mux with all the routers registered.
-func (s *Server) CreateMux(routers ...router.Router) *mux.Router {
+func (s *Server) CreateMux(ctx context.Context, routers ...router.Router) *mux.Router {
+	log.G(ctx).Debug("Registering routers")
 	m := mux.NewRouter()
-
-	log.G(context.TODO()).Debug("Registering routers")
 	for _, apiRouter := range routers {
 		for _, r := range apiRouter.Routes() {
+			if ctx.Err() != nil {
+				return m
+			}
+			log.G(ctx).WithFields(log.Fields{"method": r.Method(), "paths": []string{versionMatcher + r.Path(), r.Path()}}).Debugf("Registering %s %s", r.Method(), r.Path())
```

Comment:
```suggestion
			log.G(ctx).WithFields(log.Fields{"method": r.Method(), "paths": []string{versionMatcher + r.Path(), r.Path()}}).Debugf("Registering %s %s", r.Method(), r.Path())
```

I think we can drop the method and path from the log message now? Otherwise we end up with the same information being outputted twice (log fields AND log message).

Reply:
Yeah, I was going back and forth a bit; was wondering  if the "single message" would still be useful to have a quick glance  when reading the logs, wdyt?

Reply:
Gave it a quick try (not pushed) to see what it would look like. Here's the before / after;

<details>

```console
DEBU[2024-12-09T12:02:54.658131546Z] Registering routers
DEBU[2024-12-09T12:02:54.658146921Z] Registering GET /containers/{name:.*}/checkpoints  method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/checkpoints /containers/{name:.*}/checkpoints]"
DEBU[2024-12-09T12:02:54.658212629Z] Registering POST /containers/{name:.*}/checkpoints  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/checkpoints /containers/{name:.*}/checkpoints]"
DEBU[2024-12-09T12:02:54.658250421Z] Registering DELETE /containers/{name}/checkpoints/{checkpoint}  method=DELETE paths="[/v{version:[0-9.]+}/containers/{name}/checkpoints/{checkpoint} /containers/{name}/checkpoints/{checkpoint}]"
DEBU[2024-12-09T12:02:54.658316838Z] Registering HEAD /containers/{name:.*}/archive  method=HEAD paths="[/v{version:[0-9.]+}/containers/{name:.*}/archive /containers/{name:.*}/archive]"
DEBU[2024-12-09T12:02:54.658350254Z] Registering GET /containers/json              method=GET paths="[/v{version:[0-9.]+}/containers/json /containers/json]"
DEBU[2024-12-09T12:02:54.658378338Z] Registering GET /containers/{name:.*}/export  method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/export /containers/{name:.*}/export]"
DEBU[2024-12-09T12:02:54.658411296Z] Registering GET /containers/{name:.*}/changes  method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/changes /containers/{name:.*}/changes]"
DEBU[2024-12-09T12:02:54.658444004Z] Registering GET /containers/{name:.*}/json    method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/json /containers/{name:.*}/json]"
DEBU[2024-12-09T12:02:54.658489088Z] Registering GET /containers/{name:.*}/top     method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/top /containers/{name:.*}/top]"
DEBU[2024-12-09T12:02:54.658527463Z] Registering GET /containers/{name:.*}/logs    method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/logs /containers/{name:.*}/logs]"
DEBU[2024-12-09T12:02:54.658559796Z] Registering GET /containers/{name:.*}/stats   method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/stats /containers/{name:.*}/stats]"
DEBU[2024-12-09T12:02:54.658586588Z] Registering GET /containers/{name:.*}/attach/ws  method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/attach/ws /containers/{name:.*}/attach/ws]"
DEBU[2024-12-09T12:02:54.658613796Z] Registering GET /exec/{id:.*}/json            method=GET paths="[/v{version:[0-9.]+}/exec/{id:.*}/json /exec/{id:.*}/json]"
DEBU[2024-12-09T12:02:54.658645504Z] Registering GET /containers/{name:.*}/archive  method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/archive /containers/{name:.*}/archive]"
DEBU[2024-12-09T12:02:54.658670838Z] Registering POST /containers/create           method=POST paths="[/v{version:[0-9.]+}/containers/create /containers/create]"
DEBU[2024-12-09T12:02:54.658705838Z] Registering POST /containers/{name:.*}/kill   method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/kill /containers/{name:.*}/kill]"
DEBU[2024-12-09T12:02:54.658734629Z] Registering POST /containers/{name:.*}/pause  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/pause /containers/{name:.*}/pause]"
DEBU[2024-12-09T12:02:54.658763588Z] Registering POST /containers/{name:.*}/unpause  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/unpause /containers/{name:.*}/unpause]"
DEBU[2024-12-09T12:02:54.658790588Z] Registering POST /containers/{name:.*}/restart  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/restart /containers/{name:.*}/restart]"
DEBU[2024-12-09T12:02:54.658834296Z] Registering POST /containers/{name:.*}/start  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/start /containers/{name:.*}/start]"
DEBU[2024-12-09T12:02:54.658866629Z] Registering POST /containers/{name:.*}/stop   method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/stop /containers/{name:.*}/stop]"
DEBU[2024-12-09T12:02:54.658890629Z] Registering POST /containers/{name:.*}/wait   method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/wait /containers/{name:.*}/wait]"
DEBU[2024-12-09T12:02:54.658921754Z] Registering POST /containers/{name:.*}/resize  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/resize /containers/{name:.*}/resize]"
DEBU[2024-12-09T12:02:54.658946838Z] Registering POST /containers/{name:.*}/attach  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/attach /containers/{name:.*}/attach]"
DEBU[2024-12-09T12:02:54.658971879Z] Registering POST /containers/{name:.*}/exec   method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/exec /containers/{name:.*}/exec]"
DEBU[2024-12-09T12:02:54.658995879Z] Registering POST /exec/{name:.*}/start        method=POST paths="[/v{version:[0-9.]+}/exec/{name:.*}/start /exec/{name:.*}/start]"
DEBU[2024-12-09T12:02:54.659023379Z] Registering POST /exec/{name:.*}/resize       method=POST paths="[/v{version:[0-9.]+}/exec/{name:.*}/resize /exec/{name:.*}/resize]"
DEBU[2024-12-09T12:02:54.659046963Z] Registering POST /containers/{name:.*}/rename  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/rename /containers/{name:.*}/rename]"
DEBU[2024-12-09T12:02:54.659069588Z] Registering POST /containers/{name:.*}/update  method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/update /containers/{name:.*}/update]"
DEBU[2024-12-09T12:02:54.659091838Z] Registering POST /containers/prune            method=POST paths="[/v{version:[0-9.]+}/containers/prune /containers/prune]"
DEBU[2024-12-09T12:02:54.659109421Z] Registering POST /commit                      method=POST paths="[/v{version:[0-9.]+}/commit /commit]"
DEBU[2024-12-09T12:02:54.659126254Z] Registering PUT /containers/{name:.*}/archive  method=PUT paths="[/v{version:[0-9.]+}/containers/{name:.*}/archive /containers/{name:.*}/archive]"
DEBU[2024-12-09T12:02:54.659147879Z] Registering DELETE /containers/{name:.*}      method=DELETE paths="[/v{version:[0-9.]+}/containers/{name:.*} /containers/{name:.*}]"
DEBU[2024-12-09T12:02:54.659173338Z] Registering GET /images/json                  method=GET paths="[/v{version:[0-9.]+}/images/json /images/json]"
DEBU[2024-12-09T12:02:54.659195546Z] Registering GET /images/search                method=GET paths="[/v{version:[0-9.]+}/images/search /images/search]"
DEBU[2024-12-09T12:02:54.659214296Z] Registering GET /images/get                   method=GET paths="[/v{version:[0-9.]+}/images/get /images/get]"
DEBU[2024-12-09T12:02:54.659228046Z] Registering GET /images/{name:.*}/get         method=GET paths="[/v{version:[0-9.]+}/images/{name:.*}/get /images/{name:.*}/get]"
DEBU[2024-12-09T12:02:54.659254088Z] Registering GET /images/{name:.*}/history     method=GET paths="[/v{version:[0-9.]+}/images/{name:.*}/history /images/{name:.*}/history]"
DEBU[2024-12-09T12:02:54.659295671Z] Registering GET /images/{name:.*}/json        method=GET paths="[/v{version:[0-9.]+}/images/{name:.*}/json /images/{name:.*}/json]"
DEBU[2024-12-09T12:02:54.659327379Z] Registering POST /images/load                 method=POST paths="[/v{version:[0-9.]+}/images/load /images/load]"
DEBU[2024-12-09T12:02:54.659347004Z] Registering POST /images/create               method=POST paths="[/v{version:[0-9.]+}/images/create /images/create]"
DEBU[2024-12-09T12:02:54.659389421Z] Registering POST /images/{name:.*}/push       method=POST paths="[/v{version:[0-9.]+}/images/{name:.*}/push /images/{name:.*}/push]"
DEBU[2024-12-09T12:02:54.659432421Z] Registering POST /images/{name:.*}/tag        method=POST paths="[/v{version:[0-9.]+}/images/{name:.*}/tag /images/{name:.*}/tag]"
DEBU[2024-12-09T12:02:54.659477754Z] Registering POST /images/prune                method=POST paths="[/v{version:[0-9.]+}/images/prune /images/prune]"
DEBU[2024-12-09T12:02:54.659500171Z] Registering DELETE /images/{name:.*}          method=DELETE paths="[/v{version:[0-9.]+}/images/{name:.*} /images/{name:.*}]"
DEBU[2024-12-09T12:02:54.659519963Z] Registering OPTIONS /{anyroute:.*}            method=OPTIONS paths="[/v{version:[0-9.]+}/{anyroute:.*} /{anyroute:.*}]"
DEBU[2024-12-09T12:02:54.659556213Z] Registering GET /_ping                        method=GET paths="[/v{version:[0-9.]+}/_ping /_ping]"
DEBU[2024-12-09T12:02:54.659576796Z] Registering HEAD /_ping                       method=HEAD paths="[/v{version:[0-9.]+}/_ping /_ping]"
DEBU[2024-12-09T12:02:54.659593629Z] Registering GET /events                       method=GET paths="[/v{version:[0-9.]+}/events /events]"
DEBU[2024-12-09T12:02:54.659613671Z] Registering GET /info                         method=GET paths="[/v{version:[0-9.]+}/info /info]"
DEBU[2024-12-09T12:02:54.659631754Z] Registering GET /version                      method=GET paths="[/v{version:[0-9.]+}/version /version]"
DEBU[2024-12-09T12:02:54.659666796Z] Registering GET /system/df                    method=GET paths="[/v{version:[0-9.]+}/system/df /system/df]"
DEBU[2024-12-09T12:02:54.659688171Z] Registering POST /auth                        method=POST paths="[/v{version:[0-9.]+}/auth /auth]"
DEBU[2024-12-09T12:02:54.659705629Z] Registering GET /volumes                      method=GET paths="[/v{version:[0-9.]+}/volumes /volumes]"
DEBU[2024-12-09T12:02:54.659723463Z] Registering GET /volumes/{name:.*}            method=GET paths="[/v{version:[0-9.]+}/volumes/{name:.*} /volumes/{name:.*}]"
DEBU[2024-12-09T12:02:54.660586379Z] Registering POST /volumes/create              method=POST paths="[/v{version:[0-9.]+}/volumes/create /volumes/create]"
DEBU[2024-12-09T12:02:54.660653504Z] Registering POST /volumes/prune               method=POST paths="[/v{version:[0-9.]+}/volumes/prune /volumes/prune]"
DEBU[2024-12-09T12:02:54.660716421Z] Registering PUT /volumes/{name:.*}            method=PUT paths="[/v{version:[0-9.]+}/volumes/{name:.*} /volumes/{name:.*}]"
DEBU[2024-12-09T12:02:54.660767463Z] Registering DELETE /volumes/{name:.*}         method=DELETE paths="[/v{version:[0-9.]+}/volumes/{name:.*} /volumes/{name:.*}]"
DEBU[2024-12-09T12:02:54.661008963Z] Registering POST /build                       method=POST paths="[/v{version:[0-9.]+}/build /build]"
DEBU[2024-12-09T12:02:54.661042546Z] Registering POST /build/prune                 method=POST paths="[/v{version:[0-9.]+}/build/prune /build/prune]"
DEBU[2024-12-09T12:02:54.661064963Z] Registering POST /build/cancel                method=POST paths="[/v{version:[0-9.]+}/build/cancel /build/cancel]"
DEBU[2024-12-09T12:02:54.661087338Z] Registering POST /session                     method=POST paths="[/v{version:[0-9.]+}/session /session]"
DEBU[2024-12-09T12:02:54.661108796Z] Registering POST /swarm/init                  method=POST paths="[/v{version:[0-9.]+}/swarm/init /swarm/init]"
DEBU[2024-12-09T12:02:54.661133254Z] Registering POST /swarm/join                  method=POST paths="[/v{version:[0-9.]+}/swarm/join /swarm/join]"
DEBU[2024-12-09T12:02:54.661153671Z] Registering POST /swarm/leave                 method=POST paths="[/v{version:[0-9.]+}/swarm/leave /swarm/leave]"
DEBU[2024-12-09T12:02:54.661171588Z] Registering GET /swarm                        method=GET paths="[/v{version:[0-9.]+}/swarm /swarm]"
DEBU[2024-12-09T12:02:54.661212254Z] Registering GET /swarm/unlockkey              method=GET paths="[/v{version:[0-9.]+}/swarm/unlockkey /swarm/unlockkey]"
DEBU[2024-12-09T12:02:54.661229879Z] Registering POST /swarm/update                method=POST paths="[/v{version:[0-9.]+}/swarm/update /swarm/update]"
DEBU[2024-12-09T12:02:54.661255213Z] Registering POST /swarm/unlock                method=POST paths="[/v{version:[0-9.]+}/swarm/unlock /swarm/unlock]"
DEBU[2024-12-09T12:02:54.661274004Z] Registering GET /services                     method=GET paths="[/v{version:[0-9.]+}/services /services]"
DEBU[2024-12-09T12:02:54.661316296Z] Registering GET /services/{id}                method=GET paths="[/v{version:[0-9.]+}/services/{id} /services/{id}]"
DEBU[2024-12-09T12:02:54.661354588Z] Registering POST /services/create             method=POST paths="[/v{version:[0-9.]+}/services/create /services/create]"
DEBU[2024-12-09T12:02:54.661385213Z] Registering POST /services/{id}/update        method=POST paths="[/v{version:[0-9.]+}/services/{id}/update /services/{id}/update]"
DEBU[2024-12-09T12:02:54.661413171Z] Registering DELETE /services/{id}             method=DELETE paths="[/v{version:[0-9.]+}/services/{id} /services/{id}]"
DEBU[2024-12-09T12:02:54.661436838Z] Registering GET /services/{id}/logs           method=GET paths="[/v{version:[0-9.]+}/services/{id}/logs /services/{id}/logs]"
DEBU[2024-12-09T12:02:54.661464296Z] Registering GET /nodes                        method=GET paths="[/v{version:[0-9.]+}/nodes /nodes]"
DEBU[2024-12-09T12:02:54.661483296Z] Registering GET /nodes/{id}                   method=GET paths="[/v{version:[0-9.]+}/nodes/{id} /nodes/{id}]"
DEBU[2024-12-09T12:02:54.661511879Z] Registering DELETE /nodes/{id}                method=DELETE paths="[/v{version:[0-9.]+}/nodes/{id} /nodes/{id}]"
DEBU[2024-12-09T12:02:54.661538046Z] Registering POST /nodes/{id}/update           method=POST paths="[/v{version:[0-9.]+}/nodes/{id}/update /nodes/{id}/update]"
DEBU[2024-12-09T12:02:54.661618838Z] Registering GET /tasks                        method=GET paths="[/v{version:[0-9.]+}/tasks /tasks]"
DEBU[2024-12-09T12:02:54.661641671Z] Registering GET /tasks/{id}                   method=GET paths="[/v{version:[0-9.]+}/tasks/{id} /tasks/{id}]"
DEBU[2024-12-09T12:02:54.661662254Z] Registering GET /tasks/{id}/logs              method=GET paths="[/v{version:[0-9.]+}/tasks/{id}/logs /tasks/{id}/logs]"
DEBU[2024-12-09T12:02:54.661694421Z] Registering GET /secrets                      method=GET paths="[/v{version:[0-9.]+}/secrets /secrets]"
DEBU[2024-12-09T12:02:54.661720463Z] Registering POST /secrets/create              method=POST paths="[/v{version:[0-9.]+}/secrets/create /secrets/create]"
DEBU[2024-12-09T12:02:54.661748754Z] Registering DELETE /secrets/{id}              method=DELETE paths="[/v{version:[0-9.]+}/secrets/{id} /secrets/{id}]"
DEBU[2024-12-09T12:02:54.661776088Z] Registering GET /secrets/{id}                 method=GET paths="[/v{version:[0-9.]+}/secrets/{id} /secrets/{id}]"
DEBU[2024-12-09T12:02:54.661800504Z] Registering POST /secrets/{id}/update         method=POST paths="[/v{version:[0-9.]+}/secrets/{id}/update /secrets/{id}/update]"
DEBU[2024-12-09T12:02:54.661841671Z] Registering GET /configs                      method=GET paths="[/v{version:[0-9.]+}/configs /configs]"
DEBU[2024-12-09T12:02:54.661858546Z] Registering POST /configs/create              method=POST paths="[/v{version:[0-9.]+}/configs/create /configs/create]"
DEBU[2024-12-09T12:02:54.661882879Z] Registering DELETE /configs/{id}              method=DELETE paths="[/v{version:[0-9.]+}/configs/{id} /configs/{id}]"
DEBU[2024-12-09T12:02:54.661908129Z] Registering GET /configs/{id}                 method=GET paths="[/v{version:[0-9.]+}/configs/{id} /configs/{id}]"
DEBU[2024-12-09T12:02:54.661936379Z] Registering POST /configs/{id}/update         method=POST paths="[/v{version:[0-9.]+}/configs/{id}/update /configs/{id}/update]"
DEBU[2024-12-09T12:02:54.661966671Z] Registering GET /plugins                      method=GET paths="[/v{version:[0-9.]+}/plugins /plugins]"
DEBU[2024-12-09T12:02:54.661991379Z] Registering GET /plugins/{name:.*}/json       method=GET paths="[/v{version:[0-9.]+}/plugins/{name:.*}/json /plugins/{name:.*}/json]"
DEBU[2024-12-09T12:02:54.662023629Z] Registering GET /plugins/privileges           method=GET paths="[/v{version:[0-9.]+}/plugins/privileges /plugins/privileges]"
DEBU[2024-12-09T12:02:54.662055671Z] Registering DELETE /plugins/{name:.*}         method=DELETE paths="[/v{version:[0-9.]+}/plugins/{name:.*} /plugins/{name:.*}]"
DEBU[2024-12-09T12:02:54.662092129Z] Registering POST /plugins/{name:.*}/enable    method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/enable /plugins/{name:.*}/enable]"
DEBU[2024-12-09T12:02:54.662133129Z] Registering POST /plugins/{name:.*}/disable   method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/disable /plugins/{name:.*}/disable]"
DEBU[2024-12-09T12:02:54.662160671Z] Registering POST /plugins/pull                method=POST paths="[/v{version:[0-9.]+}/plugins/pull /plugins/pull]"
DEBU[2024-12-09T12:02:54.662183296Z] Registering POST /plugins/{name:.*}/push      method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/push /plugins/{name:.*}/push]"
DEBU[2024-12-09T12:02:54.662215463Z] Registering POST /plugins/{name:.*}/upgrade   method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/upgrade /plugins/{name:.*}/upgrade]"
DEBU[2024-12-09T12:02:54.662242504Z] Registering POST /plugins/{name:.*}/set       method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/set /plugins/{name:.*}/set]"
DEBU[2024-12-09T12:02:54.662297963Z] Registering POST /plugins/create              method=POST paths="[/v{version:[0-9.]+}/plugins/create /plugins/create]"
DEBU[2024-12-09T12:02:54.662327254Z] Registering GET /distribution/{name:.*}/json  method=GET paths="[/v{version:[0-9.]+}/distribution/{name:.*}/json /distribution/{name:.*}/json]"
DEBU[2024-12-09T12:02:54.662364379Z] Registering GET /networks                     method=GET paths="[/v{version:[0-9.]+}/networks /networks]"
DEBU[2024-12-09T12:02:54.662384296Z] Registering GET /networks/                    method=GET paths="[/v{version:[0-9.]+}/networks/ /networks/]"
DEBU[2024-12-09T12:02:54.662412296Z] Registering GET /networks/{id:.+}             method=GET paths="[/v{version:[0-9.]+}/networks/{id:.+} /networks/{id:.+}]"
DEBU[2024-12-09T12:02:54.662439546Z] Registering POST /networks/create             method=POST paths="[/v{version:[0-9.]+}/networks/create /networks/create]"
DEBU[2024-12-09T12:02:54.662460171Z] Registering POST /networks/{id:.*}/connect    method=POST paths="[/v{version:[0-9.]+}/networks/{id:.*}/connect /networks/{id:.*}/connect]"
DEBU[2024-12-09T12:02:54.662487088Z] Registering POST /networks/{id:.*}/disconnect  method=POST paths="[/v{version:[0-9.]+}/networks/{id:.*}/disconnect /networks/{id:.*}/disconnect]"
DEBU[2024-12-09T12:02:54.662518963Z] Registering POST /networks/prune              method=POST paths="[/v{version:[0-9.]+}/networks/prune /networks/prune]"
DEBU[2024-12-09T12:02:54.662548629Z] Registering DELETE /networks/{id:.*}          method=DELETE paths="[/v{version:[0-9.]+}/networks/{id:.*} /networks/{id:.*}]"
DEBU[2024-12-09T12:02:54.662578088Z] Registering GET /debug/vars                   method=GET paths="[/v{version:[0-9.]+}/debug/vars /debug/vars]"
DEBU[2024-12-09T12:02:54.662600088Z] Registering GET /debug/pprof/                 method=GET paths="[/v{version:[0-9.]+}/debug/pprof/ /debug/pprof/]"
DEBU[2024-12-09T12:02:54.662620629Z] Registering GET /debug/pprof/cmdline          method=GET paths="[/v{version:[0-9.]+}/debug/pprof/cmdline /debug/pprof/cmdline]"
DEBU[2024-12-09T12:02:54.662641088Z] Registering GET /debug/pprof/profile          method=GET paths="[/v{version:[0-9.]+}/debug/pprof/profile /debug/pprof/profile]"
DEBU[2024-12-09T12:02:54.662672004Z] Registering GET /debug/pprof/symbol           method=GET paths="[/v{version:[0-9.]+}/debug/pprof/symbol /debug/pprof/symbol]"
DEBU[2024-12-09T12:02:54.662707379Z] Registering GET /debug/pprof/trace            method=GET paths="[/v{version:[0-9.]+}/debug/pprof/trace /debug/pprof/trace]"
DEBU[2024-12-09T12:02:54.662734754Z] Registering GET /debug/pprof/{name}           method=GET paths="[/v{version:[0-9.]+}/debug/pprof/{name} /debug/pprof/{name}]"
DEBU[2024-12-09T12:02:54.662760879Z] Registering POST /grpc                        method=POST paths="[/v{version:[0-9.]+}/grpc /grpc]"
INFO[2024-12-09T12:02:54.662979754Z] API listen on /var/run/docker.sock
```

</details>


<details>

```console
DEBU[2024-12-09T12:01:33.369827925Z] Registering routers
DEBU[2024-12-09T12:01:33.369843883Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/checkpoints /containers/{name:.*}/checkpoints]"
DEBU[2024-12-09T12:01:33.369902425Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/checkpoints /containers/{name:.*}/checkpoints]"
DEBU[2024-12-09T12:01:33.369952175Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/containers/{name}/checkpoints/{checkpoint} /containers/{name}/checkpoints/{checkpoint}]"
DEBU[2024-12-09T12:01:33.370021300Z] Registering route                             method=HEAD paths="[/v{version:[0-9.]+}/containers/{name:.*}/archive /containers/{name:.*}/archive]"
DEBU[2024-12-09T12:01:33.370057133Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/json /containers/json]"
DEBU[2024-12-09T12:01:33.370078383Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/export /containers/{name:.*}/export]"
DEBU[2024-12-09T12:01:33.370114550Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/changes /containers/{name:.*}/changes]"
DEBU[2024-12-09T12:01:33.370152842Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/json /containers/{name:.*}/json]"
DEBU[2024-12-09T12:01:33.370195633Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/top /containers/{name:.*}/top]"
DEBU[2024-12-09T12:01:33.370224758Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/logs /containers/{name:.*}/logs]"
DEBU[2024-12-09T12:01:33.370254883Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/stats /containers/{name:.*}/stats]"
DEBU[2024-12-09T12:01:33.370449717Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/attach/ws /containers/{name:.*}/attach/ws]"
DEBU[2024-12-09T12:01:33.370633633Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/exec/{id:.*}/json /exec/{id:.*}/json]"
DEBU[2024-12-09T12:01:33.370693092Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/containers/{name:.*}/archive /containers/{name:.*}/archive]"
DEBU[2024-12-09T12:01:33.370779550Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/create /containers/create]"
DEBU[2024-12-09T12:01:33.370819883Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/kill /containers/{name:.*}/kill]"
DEBU[2024-12-09T12:01:33.370859383Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/pause /containers/{name:.*}/pause]"
DEBU[2024-12-09T12:01:33.371143967Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/unpause /containers/{name:.*}/unpause]"
DEBU[2024-12-09T12:01:33.371483175Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/restart /containers/{name:.*}/restart]"
DEBU[2024-12-09T12:01:33.371774383Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/start /containers/{name:.*}/start]"
DEBU[2024-12-09T12:01:33.371914008Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/stop /containers/{name:.*}/stop]"
DEBU[2024-12-09T12:01:33.371994717Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/wait /containers/{name:.*}/wait]"
DEBU[2024-12-09T12:01:33.372395675Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/resize /containers/{name:.*}/resize]"
DEBU[2024-12-09T12:01:33.372436425Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/attach /containers/{name:.*}/attach]"
DEBU[2024-12-09T12:01:33.372465300Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/exec /containers/{name:.*}/exec]"
DEBU[2024-12-09T12:01:33.372493508Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/exec/{name:.*}/start /exec/{name:.*}/start]"
DEBU[2024-12-09T12:01:33.372517383Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/exec/{name:.*}/resize /exec/{name:.*}/resize]"
DEBU[2024-12-09T12:01:33.372539217Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/rename /containers/{name:.*}/rename]"
DEBU[2024-12-09T12:01:33.372637758Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/{name:.*}/update /containers/{name:.*}/update]"
DEBU[2024-12-09T12:01:33.372670758Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/containers/prune /containers/prune]"
DEBU[2024-12-09T12:01:33.372692133Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/commit /commit]"
DEBU[2024-12-09T12:01:33.372725383Z] Registering route                             method=PUT paths="[/v{version:[0-9.]+}/containers/{name:.*}/archive /containers/{name:.*}/archive]"
DEBU[2024-12-09T12:01:33.372771592Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/containers/{name:.*} /containers/{name:.*}]"
DEBU[2024-12-09T12:01:33.372812300Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/images/json /images/json]"
DEBU[2024-12-09T12:01:33.372914008Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/images/search /images/search]"
DEBU[2024-12-09T12:01:33.372997925Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/images/get /images/get]"
DEBU[2024-12-09T12:01:33.373025008Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/images/{name:.*}/get /images/{name:.*}/get]"
DEBU[2024-12-09T12:01:33.373142342Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/images/{name:.*}/history /images/{name:.*}/history]"
DEBU[2024-12-09T12:01:33.373234717Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/images/{name:.*}/json /images/{name:.*}/json]"
DEBU[2024-12-09T12:01:33.373268633Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/images/load /images/load]"
DEBU[2024-12-09T12:01:33.373320842Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/images/create /images/create]"
DEBU[2024-12-09T12:01:33.373348967Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/images/{name:.*}/push /images/{name:.*}/push]"
DEBU[2024-12-09T12:01:33.373370383Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/images/{name:.*}/tag /images/{name:.*}/tag]"
DEBU[2024-12-09T12:01:33.373400383Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/images/prune /images/prune]"
DEBU[2024-12-09T12:01:33.373423967Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/images/{name:.*} /images/{name:.*}]"
DEBU[2024-12-09T12:01:33.373452425Z] Registering route                             method=OPTIONS paths="[/v{version:[0-9.]+}/{anyroute:.*} /{anyroute:.*}]"
DEBU[2024-12-09T12:01:33.373478133Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/_ping /_ping]"
DEBU[2024-12-09T12:01:33.373492342Z] Registering route                             method=HEAD paths="[/v{version:[0-9.]+}/_ping /_ping]"
DEBU[2024-12-09T12:01:33.373513925Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/events /events]"
DEBU[2024-12-09T12:01:33.373529217Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/info /info]"
DEBU[2024-12-09T12:01:33.373545217Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/version /version]"
DEBU[2024-12-09T12:01:33.373560717Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/system/df /system/df]"
DEBU[2024-12-09T12:01:33.373578967Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/auth /auth]"
DEBU[2024-12-09T12:01:33.373595925Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/volumes /volumes]"
DEBU[2024-12-09T12:01:33.373612592Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/volumes/{name:.*} /volumes/{name:.*}]"
DEBU[2024-12-09T12:01:33.373646758Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/volumes/create /volumes/create]"
DEBU[2024-12-09T12:01:33.373668425Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/volumes/prune /volumes/prune]"
DEBU[2024-12-09T12:01:33.373694925Z] Registering route                             method=PUT paths="[/v{version:[0-9.]+}/volumes/{name:.*} /volumes/{name:.*}]"
DEBU[2024-12-09T12:01:33.373721675Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/volumes/{name:.*} /volumes/{name:.*}]"
DEBU[2024-12-09T12:01:33.373764258Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/build /build]"
DEBU[2024-12-09T12:01:33.373781633Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/build/prune /build/prune]"
DEBU[2024-12-09T12:01:33.373807717Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/build/cancel /build/cancel]"
DEBU[2024-12-09T12:01:33.373833842Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/session /session]"
DEBU[2024-12-09T12:01:33.373850383Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/swarm/init /swarm/init]"
DEBU[2024-12-09T12:01:33.373875008Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/swarm/join /swarm/join]"
DEBU[2024-12-09T12:01:33.373890842Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/swarm/leave /swarm/leave]"
DEBU[2024-12-09T12:01:33.373909800Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/swarm /swarm]"
DEBU[2024-12-09T12:01:33.373926217Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/swarm/unlockkey /swarm/unlockkey]"
DEBU[2024-12-09T12:01:33.373946383Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/swarm/update /swarm/update]"
DEBU[2024-12-09T12:01:33.373970592Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/swarm/unlock /swarm/unlock]"
DEBU[2024-12-09T12:01:33.373989967Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/services /services]"
DEBU[2024-12-09T12:01:33.374011883Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/services/{id} /services/{id}]"
DEBU[2024-12-09T12:01:33.374036383Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/services/create /services/create]"
DEBU[2024-12-09T12:01:33.374061300Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/services/{id}/update /services/{id}/update]"
DEBU[2024-12-09T12:01:33.374092342Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/services/{id} /services/{id}]"
DEBU[2024-12-09T12:01:33.374119758Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/services/{id}/logs /services/{id}/logs]"
DEBU[2024-12-09T12:01:33.374147800Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/nodes /nodes]"
DEBU[2024-12-09T12:01:33.374165758Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/nodes/{id} /nodes/{id}]"
DEBU[2024-12-09T12:01:33.374184633Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/nodes/{id} /nodes/{id}]"
DEBU[2024-12-09T12:01:33.374211717Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/nodes/{id}/update /nodes/{id}/update]"
DEBU[2024-12-09T12:01:33.374270592Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/tasks /tasks]"
DEBU[2024-12-09T12:01:33.374360883Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/tasks/{id} /tasks/{id}]"
DEBU[2024-12-09T12:01:33.374404758Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/tasks/{id}/logs /tasks/{id}/logs]"
DEBU[2024-12-09T12:01:33.374492842Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/secrets /secrets]"
DEBU[2024-12-09T12:01:33.374522300Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/secrets/create /secrets/create]"
DEBU[2024-12-09T12:01:33.374626800Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/secrets/{id} /secrets/{id}]"
DEBU[2024-12-09T12:01:33.374703008Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/secrets/{id} /secrets/{id}]"
DEBU[2024-12-09T12:01:33.374819050Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/secrets/{id}/update /secrets/{id}/update]"
DEBU[2024-12-09T12:01:33.374846758Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/configs /configs]"
DEBU[2024-12-09T12:01:33.374864633Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/configs/create /configs/create]"
DEBU[2024-12-09T12:01:33.374884300Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/configs/{id} /configs/{id}]"
DEBU[2024-12-09T12:01:33.374907258Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/configs/{id} /configs/{id}]"
DEBU[2024-12-09T12:01:33.374928050Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/configs/{id}/update /configs/{id}/update]"
DEBU[2024-12-09T12:01:33.374958133Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/plugins /plugins]"
DEBU[2024-12-09T12:01:33.374975633Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/plugins/{name:.*}/json /plugins/{name:.*}/json]"
DEBU[2024-12-09T12:01:33.376730300Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/plugins/privileges /plugins/privileges]"
DEBU[2024-12-09T12:01:33.376828092Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/plugins/{name:.*} /plugins/{name:.*}]"
DEBU[2024-12-09T12:01:33.376861050Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/enable /plugins/{name:.*}/enable]"
DEBU[2024-12-09T12:01:33.376896217Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/disable /plugins/{name:.*}/disable]"
DEBU[2024-12-09T12:01:33.376937800Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/plugins/pull /plugins/pull]"
DEBU[2024-12-09T12:01:33.376952675Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/push /plugins/{name:.*}/push]"
DEBU[2024-12-09T12:01:33.376979425Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/upgrade /plugins/{name:.*}/upgrade]"
DEBU[2024-12-09T12:01:33.377002758Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/plugins/{name:.*}/set /plugins/{name:.*}/set]"
DEBU[2024-12-09T12:01:33.377045217Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/plugins/create /plugins/create]"
DEBU[2024-12-09T12:01:33.377084675Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/distribution/{name:.*}/json /distribution/{name:.*}/json]"
DEBU[2024-12-09T12:01:33.377110008Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/networks /networks]"
DEBU[2024-12-09T12:01:33.377128717Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/networks/ /networks/]"
DEBU[2024-12-09T12:01:33.377147550Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/networks/{id:.+} /networks/{id:.+}]"
DEBU[2024-12-09T12:01:33.377171675Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/networks/create /networks/create]"
DEBU[2024-12-09T12:01:33.377194675Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/networks/{id:.*}/connect /networks/{id:.*}/connect]"
DEBU[2024-12-09T12:01:33.377219550Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/networks/{id:.*}/disconnect /networks/{id:.*}/disconnect]"
DEBU[2024-12-09T12:01:33.377243925Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/networks/prune /networks/prune]"
DEBU[2024-12-09T12:01:33.377263300Z] Registering route                             method=DELETE paths="[/v{version:[0-9.]+}/networks/{id:.*} /networks/{id:.*}]"
DEBU[2024-12-09T12:01:33.377286675Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/debug/vars /debug/vars]"
DEBU[2024-12-09T12:01:33.377303592Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/debug/pprof/ /debug/pprof/]"
DEBU[2024-12-09T12:01:33.377325675Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/debug/pprof/cmdline /debug/pprof/cmdline]"
DEBU[2024-12-09T12:01:33.377356425Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/debug/pprof/profile /debug/pprof/profile]"
DEBU[2024-12-09T12:01:33.377377133Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/debug/pprof/symbol /debug/pprof/symbol]"
DEBU[2024-12-09T12:01:33.377398425Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/debug/pprof/trace /debug/pprof/trace]"
DEBU[2024-12-09T12:01:33.377416508Z] Registering route                             method=GET paths="[/v{version:[0-9.]+}/debug/pprof/{name} /debug/pprof/{name}]"
DEBU[2024-12-09T12:01:33.377438092Z] Registering route                             method=POST paths="[/v{version:[0-9.]+}/grpc /grpc]"
INFO[2024-12-09T12:01:33.377670342Z] API listen on /var/run/docker.sock
```

</details>

Reply:
The "post change" logs look good to me!

Reply:
OK that's 2 folks for changing it; let me update 👍

Reply:
The repeated `[/v{version:[0-9.]+}/` for each endpoint is IMO too mouthful though 🙈 

Is there any value in including the versioned path in the log?

Reply:
Mostly to log what we're _actually_ adding, instead of shoving it under the rug

Reply:
I could change it back to a string (`path`) and only a single variant (either the versioned or non-versioned one)

Reply:
IMO just a non-versioned path would be fine; maybe we could have a separate message that indicates that the endpoints below are available under versioned endpoints, so we don't repeat it for every route?

Reply:
Updated; changed it to `path` and only include the non-versioned one.